boards/common/nrf52xxxdk/include/periph_conf_common.h

```diff
@@ -34,16 +34,18 @@ extern "C" {
  * @name    SPI configuration
  * @{
  */
+#ifndef SPI_NUMOF /* <-- Allow overwriting SPI_NUMOF from periph_conf */
```

Comment:
What about moving this spi config in its own default include file in common nrf52 directory (like cfg_rtt_default.h) ?
Only ruuvitag and the other board would include it in their `periph_conf.h` and you can remove it from this file. And then no to define the structure before including periph_conf_common in nrf52dk/nrf52840dk.

Do this makes sense for you @maribu ?